homeassistant/components/hassio/auth.py

```diff
@@ -75,3 +83,46 @@ def _get_provider(self):
             await provider.async_validate_login(username, password)
         except HomeAssistantError:
             raise HTTPForbidden() from None
+
+
+class HassIOPasswordReset(HomeAssistantView):
+    """Hass.io view to handle password reset requests."""
+
+    name = "api:hassio:auth:password:reset"
+    url = "/api/hassio_auth/password_reset"
+
+    def __init__(self, hass):
+        """Initialize WebView."""
+        self.hass = hass
+
+    @RequestDataValidator(SCHEMA_API_PASSWORD_RESET)
+    async def post(self, request, data):
+        """Handle new discovery requests."""
+        hassio_ip = os.environ["HASSIO"].split(":")[0]
+        if request[KEY_REAL_IP] != ip_address(hassio_ip):
```

Comment:
Can we make this a decorator, assuming you used it before? That way there is 1 place with auth check logic that we can test.